src/main/java/spoon/support/gui/SpoonModelTree.java

```diff
@@ -284,6 +305,46 @@ public DefaultMutableTreeNode search() {
 		return null;
 	}
 
+	public DefaultMutableTreeNode expandAll(final DefaultMutableTreeNode node) {
+		if (node == null || node.isLeaf()) {
+			return null;
+		}
+		final ExecutorService executor = Executors.newSingleThreadExecutor();
+		executor.execute(new Runnable() {
+			public void run() {
+				try {
+					Queue<DefaultMutableTreeNode> q = new LinkedList<DefaultMutableTreeNode>();
+					q.add(node);
+					while (!q.isEmpty()) {
+						final DefaultMutableTreeNode n = q.poll();
```

Comment:
Creates a method `expand(node)` seems a good idea to reduce the depth of this method.